librz/core/cmd/cmd.c

```diff
@@ -5046,6 +5049,31 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(grep_stmt) {
 	rz_strbuf_free(sb);
 	char *specifier_str = rz_cmd_unescape_arg(specifier_str_es, true);
 	RZ_LOG_DEBUG("grep_stmt processed specifier: '%s'\n", specifier_str);
+	// setup cons->context->grep.icase according to cons->grep_icase
+	if (state->core->cons->grep_icase == CASE_SMART) {
+		// smartcase - when the search term is all lowercase, ignore the case,
+		// instead if the search term is uppercase or a mix, do a case-sensitive search.
+		bool has_upper = false, has_lower = false;
```

Comment:
is it really necessary to keep track of both upper and lower? I thought it would be enough to just track upper.

Reply:
It would be enough I think ... `!has_upper && !has_lower` is meaningful but is it relevant?